openapi3filter/validate_request_test.go

```diff
@@ -215,3 +216,53 @@ components:
 		})
 	}
 }
+
+func TestValidateRequest_CustomSchemaErrorMessage(t *testing.T) {
+	t.Parallel()
+
+	const spec = `
+openapi: 3.0.0
+info:
+  title: 'Validator'
+  version: 0.0.1
+paths:
+  /some:
+    post:
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                field:
+                  title: Some field
+                  type: string
+      responses:
+        '200':
+          description: Created
+`
+	router := setupTestRouter(t, spec)
+
+	req, err := http.NewRequest(http.MethodPost, "/some", strings.NewReader(`{"field":123}`))
+	require.NoError(t, err)
+
+	req.Header.Add("Content-Type", "application/json")
+
+	route, pathParams, err := router.FindRoute(req)
+	require.NoError(t, err)
+
+	validationInput := &RequestValidationInput{
+		Request:    req,
+		PathParams: pathParams,
+		Route:      route,
+		Options: &Options{
+			CustomSchemaErrorFunc: func(err *openapi3.SchemaError) string {
```

Comment:
Use the function calls to set the options slice instead of the struct.

Reply:
made by calling the `WithCustomSchemaErrorFunc` function